p2p/sql_stmt.go

```diff
@@ -48,24 +70,167 @@ func (srv *Server) initSql() error {
+		return err
+	}
+	log.Trace("Existing sql tables dropped", "database", srv.MySQLName)
+	return nil
+}
+
+func (srv *Server) createTables() error {
+	// create neighbors table
+	if _, err := srv.DB.Exec(`
+		CREATE TABLE IF NOT EXISTS neighbors (
+			node_id VARCHAR(128) NOT NULL, 
+			hash VARCHAR(64) NOT NULL, 
+			ip VARCHAR(39) NOT NULL, 
+			tcp_port SMALLINT unsigned NOT NULL, 
+			udp_port SMALLINT unsigned NOT NULL, 
+			first_received_at DECIMAL(18,6) NOT NULL, 
+			last_received_at DECIMAL(18,6) NOT NULL, 
+			count BIGINT unsigned DEFAULT 1, 
+			PRIMARY KEY (node_id, ip, tcp_port, udp_port)
+		)
+	`); err != nil {
+		log.Error("Failed to create neighbors table", "database", srv.MySQLName, "err", err)
+		return err
+	}
+	log.Trace("neighbors table already exists or is newly created", "database", srv.MySQLName)
+
+	// create node_meta_info table
+	if _, err := srv.DB.Exec(`
+		CREATE TABLE IF NOT EXISTS node_meta_info (
+			node_id VARCHAR(128) NOT NULL, 
+			hash VARCHAR(64) NOT NULL, 
+			dial_count BIGINT unsigned DEFAULT 0, 
+			accept_count BIGINT unsigned DEFAULT 0, 
+			too_many_peers_count BIGINT unsigned DEFAULT 0, 
+			PRIMARY KEY (node_id)
+		)
+	`); err != nil {
+		log.Error("Failed to create node_meta_info table", "database", srv.MySQLName, "err", err)
+		return err
+	}
+	log.Trace("node_meta_info table already exists or is newly created", "database", srv.MySQLName)
+
+	// create node_info table
+	if _, err := srv.DB.Exec(`
+		CREATE TABLE IF NOT EXISTS node_info (
+			id BIGINT unsigned NOT NULL AUTO_INCREMENT, 
+			node_id VARCHAR(128) NOT NULL, 
+			ip VARCHAR(39) NOT NULL, 
+			tcp_port SMALLINT unsigned NOT NULL, 
+			remote_port SMALLINT unsigned NOT NULL, 
+			p2p_version BIGINT unsigned NULL, 
+			client_id VARCHAR(255) NULL, 
+			caps VARCHAR(255) NULL, 
+			listen_port SMALLINT unsigned NULL, 
+			first_hello_at DECIMAL(18,6) NULL, 
+			last_hello_at DECIMAL(18,6) NULL, 
+			protocol_version BIGINT unsigned NULL, 
+			network_id BIGINT unsigned NULL, 
+			first_received_td DECIMAL(65) NULL, 
+			last_received_td DECIMAL(65) NULL, 
+			best_hash VARCHAR(64) NULL, 
+			genesis_hash VARCHAR(64) NULL, 
+			dao_fork TINYINT NULL, 
+			first_status_at DECIMAL(18,6) NULL, 
+			last_status_at DECIMAL(18,6) NULL, 
+			PRIMARY KEY (id), 
+			KEY (node_id)
+		)
+	`); err != nil {
+		log.Error("Failed to create node_info table", "database", srv.MySQLName, "err", err)
+		return err
+	}
+	log.Trace("node_info table already exists or is newly created", "database", srv.MySQLName)
+
+	return nil
+}
+
 func (srv *Server) CloseSql() {
 	if srv.DB != nil {
 		// close prepared sql statements
 		srv.closeSqlStmts()
 
 		// close db handle
-		if err := srv.DB.Close(); err != nil {
-			log.Error("Failed to close sql db handle", "database", srv.MySQLName, "err", err)
-		} else {
-			log.Trace("Closed sql db handle", "database", srv.MySQLName)
-		}
+		srv.closeDB(srv.DB)
```

Comment:
I don't get this change. You create a function that handles the err case, but for some reason that function still returns the err, but you don't check for it here?

Reply:
Yea.. I have no idea what happened here. I think I originally made the function because I thought I would need to check the err in some other places but ended up not using it anywhere else.
Should I keep the function (with no return value) or revert it back to how it was?

Reply:
You should keep the function and remove the returned error value (assuming that the error is never too serious and might require hard fail).

Reply:
done